### PR TITLE
Corrected disabled Query Param handling

### PR DESCRIPTION
**Pull request by kunagpal**
With this pull request, a query param list with one or more disabled query params will be correctly converted to it's string equivalent.

This takes care of https://github.com/postmanlabs/newman/issues/1114, https://github.com/postmanlabs/postman-app-support/issues/3361 and #415

### CODE REVIEW

**Comment by kunagpal** [comment]
@shamasis I've compacted and cleaned the logic used here.

File: lib/collection/query-param.js [hunk 1]
```diff
@@ -136,26 +136,34 @@ _.assign(QueryParam, /** @lends QueryParam */ {
...
-        str = params.map(function (param) {
-            return QueryParam.unparseSingle(param, encode);
-        }).join(AMPERSAND);
+        var encode = options && options.encode,
+            ignoreDisabled = options && options.ignoreDisabled,
+
+            // construct a query parameter string from the list, with considerations for disabled values
+            str = params.reduce(function (result, param) {
+                // If disabled parameters are to be ignored, bail out here
+                if (ignoreDisabled && param.disabled) { return result; }
```
Comment (shamasis) [inline lib/collection/query-param.js]: Disabled === true?

File: lib/collection/query-param.js [hunk 2]
```diff
@@ -136,26 +136,34 @@ _.assign(QueryParam, /** @lends QueryParam */ {
...
         }
 
-        str = params.map(function (param) {
-            return QueryParam.unparseSingle(param, encode);
-        }).join(AMPERSAND);
+        var encode = options && options.encode,
+            ignoreDisabled = options && options.ignoreDisabled,
+
+            // construct a query parameter string from the list, with considerations for disabled values
+            str = params.reduce(function (result, param) {
```
Comment (shamasis) [inline lib/collection/query-param.js]: Keep str; in var and move the function body out of var definition. One level less indentation and that the main work of this function is then not tucked inside a var.

Comment (kunagpal) [inline lib/collection/query-param.js]: Done 👍

File: lib/collection/query-param.js (hunk 1 above)
Comment (kunagpal) [inline lib/collection/query-param.js]: Done 👍

**Comment by shamasis** [comment]
ALso, new algorithm is 4.7 times faster than old.

**Pull request merged**